tests/src/analysis/testqgsprocessing.cpp

```diff
@@ -6144,8 +6153,10 @@ void TestQgsProcessing::modelExecution()
 
   // with values
   variables = model2.variablesForChildAlgorithm( "cx2", context, modelInputs, childResults );
-  QCOMPARE( variables.count(), 9 );
+  QCOMPARE( variables.count(), 11 );
   QCOMPARE( variables.value( "DIST" ).value.toInt(), 271 );
+  QCOMPARE( variables.value( "SOURCE_LAYER" ).source.parameterName(), QString( "SOURCE_LAYER" ) );
+  QCOMPARE( variables.value( "cx1_OUTPUT" ).source.parameterName(), QString( "" ) );
```

Comment:
@nyalldawson Not clear what would be the best value to test, I'm almost lost and I really need a document to give a hierarchy  (or network) of all processing classes.

Reply:
Better test here both outputChildId and outputName:

   QCOMPARE( variables.value( "cx1_OUTPUT" ).source.outputChildId(), ...
   QCOMPARE( variables.value( "cx1_OUTPUT" ).source.outputName(), ...